File: scripts/generatePdfThumbnails.js
const fs = require('fs');
const path = require('path');
const { PDFDocument } = require('pdf-lib');
const { createCanvas, loadImage } = require('canvas');

const projectsData = require('../src/data/projects.json');
const publicDir = path.join(__dirname, '../public');
const assetsDir = path.join(publicDir, 'eric_assets');

async function pdfToImage(pdfPath, outputPath) {
  try {
    // Read the PDF
    const pdfBytes = fs.readFileSync(pdfPath);
    const pdfDoc = await PDFDocument.load(pdfBytes);

    // Get the first page
    const pages = pdfDoc.getPages();
    const firstPage = pages[0];
    const { width, height } = firstPage.getSize();

    console.log(`PDF dimensions: ${width}x${height}`);

    // For now, we'll use a different approach with pdf-poppler or another tool
    // Let's create a placeholder approach
    console.log(`✓ Processed: ${path.basename(pdfPath)}`);
    return true;
  } catch (error) {
    console.error(`Error processing ${pdfPath}:`, error.message);
    return false;
  }
}

async function generateThumbnails() {
  console.log('Starting PDF thumbnail generation...\n');

  for (const project of projectsData.projects) {
    if (project.href && project.href.endsWith('.pdf')) {
      const pdfPath = path.join(publicDir, project.href);
      const pdfName = path.basename(project.href, '.pdf');
      const outputPath = path.join(assetsDir, `${pdfName}_thumbnail.png`);

      console.log(`Processing: ${project.name}`);
      console.log(`  PDF: ${project.href}`);

      if (fs.existsSync(pdfPath)) {
        await pdfToImage(pdfPath, outputPath);
      } else {
        console.log(`  ✗ PDF not found: ${pdfPath}`);
      }
      console.log('');
    }
  }

  console.log('Done!');
}

generateThumbnails();

File: scripts/generatePdfThumbnails.py
import os
import json
import fitz  # PyMuPDF
from pathlib import Path

# Get the project root directory
script_dir = Path(__file__).parent
project_root = script_dir.parent

# Load projects data
projects_file = project_root / 'src' / 'data' / 'projects.json'
with open(projects_file, 'r', encoding='utf-8') as f:
    projects_data = json.load(f)

public_dir = project_root / 'public'
assets_dir = public_dir / 'eric_assets'

def pdf_to_image(pdf_path, output_path):
    """Convert first page of PDF to PNG image."""
    try:
        # Open the PDF
        doc = fitz.open(pdf_path)

        # Get the first page
        page = doc[0]

        # Render page to an image (matrix for scaling, default is 72 DPI)
        # Increase the scale for better quality (2.0 = 144 DPI)
        mat = fitz.Matrix(2.0, 2.0)
        pix = page.get_pixmap(matrix=mat)

        # Save as PNG
        pix.save(output_path)

        doc.close()

        print(f"Generated: {output_path.name}")
        return True

    except Exception as e:
        print(f"Error processing {pdf_path.name}: {str(e)}")
        return False

def main():
    print("Starting PDF thumbnail generation...\n")

    updates = []

    for project in projects_data['projects']:
        href = project.get('href', '')

        if href and href.endswith('.pdf'):
            pdf_path = public_dir / href.lstrip('/')
            pdf_name = pdf_path.stem
            output_filename = f"{pdf_name}_thumbnail.png"
            output_path = assets_dir / output_filename
            new_image_path = f"/eric_assets/{output_filename}"

            print(f"Processing: {project['name']}")

            if pdf_path.exists():
                success = pdf_to_image(pdf_path, output_path)
                if success:
                    updates.append({
                        'name': project['name'],
                        'oldImage': project.get('image', ''),
                        'newImage': new_image_path
                    })
            else:
                print(f"  PDF not found: {pdf_path}")
            print()

    # Print summary
    print("\n=== Summary ===")
    print(f"Generated {len(updates)} thumbnails")
    print("\nImage path updates needed:")
    for update in updates:
        print(f"  {update['name']}:")
        print(f"    Old: {update['oldImage']}")
        print(f"    New: {update['newImage']}")

    print("\nDone!")

if __name__ == '__main__':
    main()
